src/graphql/entries.ts

```diff
@@ -98,7 +103,7 @@ function updateReferenceField(
   );
   // if we have the typename of the updated reference, we can return with it
   if (entityTypename) {
-    return { ...updatedReference, __typename: entityTypename };
+    return { ...updatedReference, __typename: entityTypename, ...referenceFromPreviewApp };
```

Comment:
```suggestion
    return { ...referenceFromPreviewApp, ...updatedReference, __typename: entityTypename, };
```

shouldn't it be like that? otherwise the `referenceFromPreviewApp` could overwrite updatedReference again 🤔

Reply:
can write a test

Reply:
@chrishelgert have added tests for single and multi ref to keep graphql properties